controller_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"net/http"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/set"
+	"github.com/juju/version"
+	gc "gopkg.in/check.v1"
+)
+
+type versionSuite struct {
+}
+
+var _ = gc.Suite(&versionSuite{})
+
+func (*versionSuite) TestSupportedVersions(c *gc.C) {
+	for _, apiVersion := range supportedAPIVersions {
+		_, _, err := version.ParseMajorMinor(apiVersion)
+		c.Check(err, jc.ErrorIsNil)
+	}
+}
+
+type controllerSuite struct {
+	testing.CleanupSuite
+	server *SimpleTestServer
```

Comment:
Yes, I think that would be a good idea. I'd like to get a few more parts in place first, and deal with the first PUT and POST calls before the shim design is likely to drop out.